drivers/SmartThings/matter-appliance/src/matter-microwave-oven/init.lua

```diff
@@ -0,0 +1,261 @@
+  end
+  local event = capabilities.mode.supportedModes(supported_mode, {visibility = {displayed = false}})
+  device:emit_event_for_endpoint(device.MATTER_DEFAULT_ENDPOINT, event)
+  event = capabilities.mode.supportedArguments(supported_mode, {visibility = {displayed = false}})
+  device:emit_event_for_endpoint(device.MATTER_DEFAULT_ENDPOINT, event)
+end
+
+local function operational_error_attr_handler(driver, device, ib, response)
+  if version.api < 10 then
+    clusters.OperationalState.types.ErrorStateStruct:augment_type(ib.data)
+  end
+  log.info_with({ hub_logs = true },
+    string.format("operational_error_attr_handler errorStateID: %s", ib.data.elements.error_state_id.value))
+  local operationalError = ib.data.elements.error_state_id.value
+  if operationalError == clusters.OperationalState.types.ErrorStateEnum.UNABLE_TO_START_OR_RESUME then
+    device:emit_event_for_endpoint(ib.endpoint_id, capabilities.operationalState.operationalState.unableToStartOrResume())
+  elseif operationalError == clusters.OperationalState.types.ErrorStateEnum.UNABLE_TO_COMPLETE_OPERATION then
+    device:emit_event_for_endpoint(ib.endpoint_id, capabilities.operationalState.operationalState.unableToCompleteOperation())
+  elseif operationalError == clusters.OperationalState.types.ErrorStateEnum.COMMAND_INVALID_IN_STATE then
+    device:emit_event_for_endpoint(ib.endpoint_id, capabilities.operationalState.operationalState.commandInvalidInCurrentState())
+  end
+  if operationalError ~= clusters.OperationalState.types.ErrorStateEnum.NO_ERROR then
+    local event = capabilities.mode.supportedModes({}, {visibility = {displayed = false}})
+    device:emit_event_for_endpoint(device.MATTER_DEFAULT_ENDPOINT, event)
+    event = capabilities.mode.supportedArguments({}, {visibility = {displayed = false}})
+    device:emit_event_for_endpoint(device.MATTER_DEFAULT_ENDPOINT, event)
+  end
+end
+
+local function microwave_oven_supported_modes_handler(driver, device, ib, response)
+  local microwaveOvenModeSupportedModes = {}
+  for _, mode in ipairs(ib.data.elements) do
+    if version.api < 11 then
+      clusters.MicrowaveOvenMode.types.ModeOptionStruct:augment_type(mode)
+    end
+    log.info_with({hub_logs=true},"Inserting supported microwave mode:", mode.elements.label.value)
+    table.insert(microwaveOvenModeSupportedModes, mode.elements.label.value)
+  end
+  local event = capabilities.mode.supportedModes(microwaveOvenModeSupportedModes, {visibility = {displayed = false}})
+  device:emit_event_for_endpoint(device.MATTER_DEFAULT_ENDPOINT, event)
+  event = capabilities.mode.supportedArguments(microwaveOvenModeSupportedModes, {visibility = {displayed = false}})
+  device:emit_event_for_endpoint(device.MATTER_DEFAULT_ENDPOINT, event)
+  device:set_field(MICROWAVE_OVEN_SUPPORTED_MODES_KEY, microwaveOvenModeSupportedModes)
+end
+
+local function microwave_oven_current_mode_handler(driver, device, ib, response)
+  log.info_with({ hub_logs = true },
```

Comment:
Remove this log or keep it in the device logs if needed, but it should be in the hub logs

Reply:
I'll fix it